### PR TITLE
Update more BBEs

### PR DESCRIPTION
**Pull request by MaryamZi**
Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: examples/clone/clone.bal [hunk 1]
```diff
@@ -35,30 +35,14 @@ public function main() {
...
-        io:println("Cloned value: ", result);
-        // Checks reference inequality of the original value and the cloned value.
-        io:println("Source and Clone are at two different memory locations: ", result !== person);
-    // If the result is an error, prints the detailed error message.
-    } else if (result is error) {
-        io:println("Cannot clone: ", result.detail().message);
-    }
+    io:println("Source value: ", person);
+    io:println("Cloned value: ", result);
+    // Check reference inequality of the original value and the cloned value.
```
Comment (praneesha) [inline examples/clone/clone.bal]:
```suggestion
    // Check reference inequality between the original value and the cloned value.
```

File: examples/immutable-values/immutable_values.bal [hunk 2]
```diff
@@ -1,91 +1,50 @@
...
 
-    // Freezes the map `m1` and assigns the returned value to another variable.
-    // Since the map is of type `anydata`, the return value would be of the same type as `m1`.
-    map<string|int> m2 = m1.freeze();
+    // Call `.cloneReadOnly()` on the map `m1` and assign the returned value to another variable.
+    map<string|int> m2 = m1.cloneReadOnly();
 
-    // Reference equality checks for `m1` and `m2` should evaluate to true, since the same value is returned with the
-    // frozen flag set.
+    // Reference equality checks for `m1` and `m2` should evaluate to false, due to a clone being created
```
Comment (praneesha) [inline examples/immutable-values/immutable_values.bal]:
```suggestion
    // Reference equality checks for `m1` and `m2` should evaluate to false due to a clone being created
```

File: examples/immutable-values/immutable_values.bal [hunk 3]
```diff
@@ -1,91 +1,50 @@
...
     io:println("m1 === m2: ", m1 === m2);
 
-    // Checks if `m1` is frozen.
-    io:println("Frozen status of m1: ", m1.isFrozen());
+    // Check if `m1` is immutable. This evaluates to false since no changes are done to
+    // the original value.
+    io:println("m1 is immutable: ", m1.isReadOnly());
+
+    // Check if `m2` is immutable. This evaluates to true since the returned clone is
+    // marked immutable.
```
Comment (praneesha) [inline examples/immutable-values/immutable_values.bal]:
```suggestion
    // marked as immutable.
```

File: examples/immutable-values/immutable_values.bal [hunk 4]
```diff
@@ -1,91 +1,50 @@
...
+    io:println("m2 is immutable: ", m2.isReadOnly());
 
-    // Attempts to add an entry to the `map`, and `trap` the panic if it results in a panic.
+    // Attempt to add an entry to the `map`, and trap the panic if it results in a panic.
```
Comment (praneesha) [inline examples/immutable-values/immutable_values.bal]:
```suggestion
    // Attempt to add an entry to the `map` and trap the panic if it results in a panic.
```

File: examples/immutable-values/immutable_values.bal [hunk 5]
```diff
@@ -1,91 +1,50 @@
...
-
-    // Attempts to freeze `m4`.
-    freezeResult = m4.freeze();
-    if (freezeResult is error) {
-        io:println("'.freeze()' failed for m4: ",
-                   <string>freezeResult.detail().message);
-    } else {
-        io:println("'.freeze()' successful for m4");
-    }
+    // Reference equality checks for `m2` and `m3` should evaluate to true, due to no clones being created
```
Comment (praneesha) [inline examples/immutable-values/immutable_values.bal]:
```suggestion
    // Reference equality checks for `m2` and `m3` should evaluate to true due to no clones being created
```

File: examples/immutable-values/immutable_values.bal [hunk 6]
```diff
@@ -1,91 +1,50 @@
...
     // values of the type `string` only.
     map<string|int> m5 = { valueType: "map", constraint: "string" };
-    // Freezes the `map`. The `.freeze()` attempt will be successful
-    // since the constraint is `anydata`. The frozen `map` only
-    // contains values of the type `string`.
-    var frozenVal = m5.freeze();
-    // Checking if the frozen value is of the type `map<string>` would
-    // evaluate to `true`.
+    // Make the map immutable. The resultant value would only
+    // contains values of the type `string`, and no values can now be
```
Comment (praneesha) [inline examples/immutable-values/immutable_values.bal]:
```suggestion
    // contain values of the type `string` and no values can now be
```

File: examples/immutable-values/immutable_values.bal [hunk 7]
```diff
@@ -1,91 +1,50 @@
...
+    // added to the map.
+    var frozenVal = m5.cloneReadOnly();
+    // Checking if the frozen value is of the type `map<string>` thus
```
Comment (praneesha) [inline examples/immutable-values/immutable_values.bal]:
```suggestion
    // Checkif the frozen value is of the type `map<string>`. Thus, it
```

File: examples/immutable-values/immutable_values.description [hunk 8]
```diff
@@ -1,12 +1,11 @@
-// Ballerina allows making values immutable or read-only.
+// Ballerina allows creating immutable or read-only clones of values.
 //
-// A value of the type `anydata` can be made immutable by calling the `.freeze()` built-in method on it. A successful
-// freeze attempt would return the value with the frozen flag set. Calling freeze on structures would deeply
-// freeze all elements.
+// The `.cloneReadOnly()` method can be called on a value of type `anydata` to create a clone of the value
```
Comment (praneesha) [inline examples/immutable-values/immutable_values.description]:
```suggestion
// The `.cloneReadOnly()` method can be called on a value of type `anydata` to create a clone of the value,
```

File: examples/json-arrays/json_arrays.bal [hunk 9]
```diff
@@ -1,35 +1,46 @@
...
+    // JSON array elements can be accessed by index.
+    // The `json` value `j1` first needs to be cast to a `json[]` to use member access.
+    // Alternatively, `j1` could have been defined as a `json[]`.
+    json[] j2 = <json[]> j1;
+    json j3 = j2[4];
+    io:println(j3);
 
-    // Adds or changes the elements in a JSON array.
-    j1[4] = 8.00;
+    // Similarly, member access could be used with `json[]` typed variables to add or
```
Comment (praneesha) [inline examples/json-arrays/json_arrays.bal]:
```suggestion
    // Similarly, member access could be used with `json[]`-typed variables to add or
```

File: examples/json-objects/json_objects.bal [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    // Create an empty JSON object. This is equivalent to an empty `map<json>`.
+    json empty = {};
+
+    int age = 30;
+    // Create a JSON object. Keys can be defined with or without quotes.
+    // Values can be any `json` expression.
+    json p = { fname: "John", lname: "Stallone", "age": age };
+    io:println(p);
+
+    // You can add or change JSON object values, using member access (`[expr]` operator).
```
Comment (praneesha) [inline examples/json-objects/json_objects.bal]:
```suggestion
    // You can add or change JSON object values using member access (i.e., the `[expr]` operator).
```

File: examples/json-objects/json_objects.bal [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+    // In order to change fields of an object value, it needs to be accessed
+    // as a `map<json>`. This is done by casting `p` to `map<json>`.
+    // Alternatively `p` could have been defined as a `map<json>`.
```
Comment (praneesha) [inline examples/json-objects/json_objects.bal]:
```suggestion
    // Alternatively, `p` could have been defined as a `map<json>`.
```

File: examples/json-objects/json_objects.bal [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+    map<json> mp = <map<json>> p;
+    mp["age"] = 31;
+    io:println(p);
+
+    // Create a Nested JSON object.
```
Comment (praneesha) [inline examples/json-objects/json_objects.bal]:
```suggestion
    // Create a nested JSON object.
```

File: examples/json-objects/tests/json_objects_test.bal [hunk 13]
```diff
@@ -0,0 +1,46 @@
+import ballerina/test;
+import ballerina/io;
+
+anydata[] outputs = [];
+int counter = 0;
+
+// This is the mock function that will replace the real function.
```
Comment (praneesha) [inline examples/json-objects/tests/json_objects_test.bal]:
```suggestion
// This is the mock function, which will replace the real function.
```

File: examples/json-objects/tests/json_objects_test.bal [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+    foreach any sa in s {
+        anydata value = <anydata> sa;
+        outputs[counter] = value.clone();
+        counter += 1;
+    }
+}
+
+@test:Config
+function testFunc() {
+    // Invoking the main function.
```
Comment (praneesha) [inline examples/json-objects/tests/json_objects_test.bal]:
```suggestion
    // Invoke the main function.
```

File: examples/json/json.bal [hunk 15]
```diff
@@ -1,68 +1,72 @@
...
 
-    // Creates a JSON Array. They are arrays including any JSON values.
+    // Create a JSON Array. This is equivalent to a `json[]`.
     json j7 = [1, false, null, "foo", { first: "John", last: "Pala" }];
     io:println(j7);
 
-    // Creates an empty JSON Object.
-    json empty = {};
+    // The `.mergeJson()` method can be used to merge two `json` values.
+    // If either of the two values is `null` the result of `.mergeJson()` is
```
Comment (praneesha) [inline examples/json/json.bal]:
```suggestion
    // If either of the two values is `null`, the result of `.mergeJson()` is
```

File: examples/json/json.bal [hunk 16]
```diff
@@ -1,68 +1,72 @@
...
+    json|error j8 = j5.mergeJson(j7);
+    io:println(j8);
 
-    int age = 30;
-    // Creates a JSON object. Keys can be defined with or without quotes.
-    // Values can be any expression.
-    json p = { fname: "John", lname: "Stallone", "age": age };
-    io:println(p);
+    // `.mergeJson()` can also be used to merge two JSON objects.
+    // Where both `json` values are mappings, for each entry in the second,
```
Comment (praneesha) [inline examples/json/json.bal]:
```suggestion
    // When both `json` values map, for each entry in the second,
```

File: examples/json/json.bal [hunk 17]
```diff
@@ -1,68 +1,72 @@
...
-         "age": age,
-         address: {
-             line: "20 Palm Grove",
-             city: "Colombo 03",
-             country: "Sri Lanka"
-         }
-    };
-    io:println(p2);
+    // The `.fromJsonString()` defined on `string` values attempts parsing the
+    // string expected to be in JSON format and returns the JSON value represented
```
Comment (praneesha) [inline examples/json/json.bal]:
```suggestion
    // string expected to be in the JSON format and returns the represented JSON value 
```

File: examples/length/length.description [hunk 18]
```diff
@@ -1,2 +1,2 @@
-// Ballerina provides a built-in `length()` method, which returns the length/size of the given value.
-// This `length()` method is supported by `string`, `array`, `map`, `record`, `table`, `json`, `xml` and `tuple` types.
+// Ballerina provides a `.length()` method for strings, arrays, tuples, maps, records, tables, and xml,
```
Comment (praneesha) [inline examples/length/length.description]:
```suggestion
// Ballerina provides a `.length()` method for strings, arrays, tuples, maps, records, tables, and XML. This returns
```

File: examples/immutable-values/immutable_values.bal (hunk 7 above)
Comment (MaryamZi) [reply examples/immutable-values/immutable_values.bal]: This change seems to introduce a disconnect here. This relates to the previous block.

File: examples/json/json.bal (hunk 16 above)
Comment (MaryamZi) [reply examples/json/json.bal]: What we are trying to say is that both of them are mappings of values.

**Pull request merged**